src/go/pt-mongodb-index-check/main.go

```diff
@@ -0,0 +1,188 @@
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/alecthomas/kong"
+	"github.com/percona/percona-toolkit/src/go/pt-mongodb-index-check/indexes"
+	"github.com/percona/percona-toolkit/src/go/pt-mongodb-index-check/templates"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type cmdlineArgs struct {
+	CheckUnused     struct{} `cmd:"" name:"check-unused" help:"Check for unused indexes."`
+	CheckDuplicated struct{} `cmd:"" name:"check-duplicates" help:"Check for duplicated indexes."`
+	CheckAll        struct{} `cmd:"" name:"check-all" help:"Check for unused and duplicated indexes."`
+	ShowHelp        struct{} `cmd:"" default:"1"`
+	Version         struct{} `cmd:"" name:"version"`
+
+	AllDatabases bool     `name:"all-databases" xor:"db" help:"Check in all databases excluding system dbs"`
+	Databases    []string `name:"databases" xor:"db" help:"Comma separated list of databases to check"`
+
+	AllCollections bool     `name:"all-collections" xor:"colls" help:"Check in all collections in the selected databases."`
+	Collections    []string `name:"collections" xor:"colls" help:"Comma separated list of collections to check"`
+	URI            string   `name:"mongodb.uri" required:"" placeholder:"mongodb://host:port/admindb?options" help:"Connection URI"`
+	JSON           bool     `name:"json" help:"Show output as JSON"`
+}
+
+type response struct {
+	Unused     []indexes.IndexStat
+	Duplicated []indexes.Duplicate
+}
+
+const (
+	TOOLNAME = "pt-mongodb-index-check"
+)
+
+var (
+	Build     string = "2020-04-23" //nolint
+	GoVersion string = "1.14.1"     //nolint
+	Version   string = "3.4.0"      //nolint
+	Commit    string                //nolint
+)
+
+func main() {
+	var args cmdlineArgs
+	kongctx := kong.Parse(&args, kong.UsageOnError())
+
+	if kongctx.Command() == "version" {
+		fmt.Println(TOOLNAME)
+		fmt.Printf("Version %s\n", Version)
+		fmt.Printf("Build: %s using %s\n", Build, GoVersion)
+		fmt.Printf("Commit: %s\n", Commit)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	if !strings.HasPrefix(args.URI, "mongodb") && !strings.HasPrefix(args.URI, "mongodb+srv") {
+		args.URI = "mongodb://" + args.URI
+	}
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(args.URI))
+	if err != nil {
+		log.Fatalf("Cannot connect to the database: %q", err)
+	}
+
+	if args.AllDatabases {
+		args.Databases, err = client.ListDatabaseNames(context.TODO(), primitive.D{})
+		if err != nil {
+			log.Fatalf("cannot list all databases: %s", err)
+		}
+	}
+	if args.AllCollections {
+		args.Collections = nil
+	}
+
+	resp := response{}
+
+	switch kongctx.Command() {
+	case "check-unused":
+		resp.Unused = append(resp.Unused, findUnused(ctx, client, args.Databases, args.Collections)...)
```

Comment:
using append here looks weird? was it done intentionally?

Reply:
No. Initially the collections loop was outside this switch and I forgot to change it.
Thanks